Sources/Datadog/Tracing/UUIDs/TracingUUIDGenerator.swift

```diff
@@ -11,9 +11,25 @@ internal protocol TracingUUIDGenerator {
 }
 
 internal struct DefaultTracingUUIDGenerator: TracingUUIDGenerator {
+    /// Lower boundary of `TracingUUID`.
+    /// `0` is reserved for historical reason: 0 == "unset", ref: dd-trace-java:DDId.java.
+    internal static let min: UInt64 = 1
+    /// Upper boundary of `TracingUUID`.
+    /// It equals to `2 ^ 63 - 1` because some tracers can't handle the `2 ^ 64 -1` range, ref: dd-trace-java:DDId.java.
+    internal static let max = UInt64.max >> 1
+
+    internal let min: UInt64
+    internal let max: UInt64
+
+    init(
+        lowerBoundary: UInt64 = DefaultTracingUUIDGenerator.min,
+        upperBoundary: UInt64 = DefaultTracingUUIDGenerator.max
+    ) {
+        self.min = lowerBoundary
+        self.max = upperBoundary
+    }
```

Comment:
we can use 1 `ClosedRange<UInt64>` instead of 2 `UInt64` instances
that way we don't need to worry about `lowerBoundary > upperBoundary` case as it's handled by `ClosedRange`

```swift
internal static let defaultRange = 1...(UInt64.max << 1) // some comments

internal let range: ClosedRange<UInt64>
init(range: ClosedRange<UInt64> = Self.defaultRange) { self.range = range }
```

Reply:
Good idea 🏅, done ✅.